webapp-django/crashstats/topcrashers/views.py

```diff
@@ -80,35 +81,40 @@ def get_topcrashers_results(**kwargs):
 
             # Number of crashes happening during garbage collection.
             hit['is_gc_count'] = 0
-
             sig_gc = hit['facets']['is_garbage_collecting']
             for row in sig_gc:
                 if row['term'].lower() == 't':
                     hit['is_gc_count'] = row['count']
 
             # Number of plugin crashes.
             hit['plugin_count'] = 0
-
             sig_process = hit['facets']['process_type']
             for row in sig_process:
                 if row['term'].lower() == 'plugin':
                     hit['plugin_count'] = row['count']
 
             # Number of hang crashes.
             hit['hang_count'] = 0
-
             sig_hang = hit['facets']['hang_type']
             for row in sig_hang:
                 # Hangs have weird values in the database: a value of 1 or -1
                 # means it is a hang, a value of 0 or missing means it is not.
                 if row['term'] in (1, -1):
                     hit['hang_count'] += row['count']
 
-            # Number of startup crashes.
-            hit['startup_percent'] = 0
-
-            sig_startup = hit['facets']['histogram_uptime']
-            for row in sig_startup:
+            # Number of crashes happening during startup. This is defined by
+            # the software, as opposed to the next method which relies on
+            # the uptime of the software.
+            hit['startup_count'] = sum(
+                row['count'] for row in hit['facets']['startup_crash']
+                if row['term'] in ('T', '1')
+            )
+
+            # Is a startup crash if more than half of the crashes are happening
+            # in the first minute after launch.
+            hit['startup_crash'] = False
+            sig_uptime = hit['facets']['histogram_uptime']
+            for row in sig_uptime:
                 if row['term'] == 0:
```

Comment:
I struggled to understand this if statement, so I debugged it and later realized this is the first histogram bucket crashes whose uptime was **less than 60 seconds**. 
I think we should have a comment here for the next schmuck getting equally confused. For example...

```python
# Only calculate the ratio based on the count of the first histogram bucket
# where the number of seconds is >= 0 and < 60 seconds. 
if row['term'] == 0:
    ...
```
or
```python
if row['term'] < 60:  # first histogram bucket
    ...
```